components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/listener/DefaultInboundUserProvisioningListener.java

```diff
@@ -230,9 +224,154 @@ public boolean doPreSetUserClaimValues(String userName, Map<String, String> inbo
+                    outboundAttributes);
+
+            Map<String, String> inboundAttributes = new HashMap<>();
+            for(int i =0; i<attributesToDelete.length;i++){
+                inboundAttributes.put(attributesToDelete[i],"");
+            }
+            ;
+            // set the in-bound attribute list.
+            provisioningEntity.setInboundAttributes(inboundAttributes);
+
+            String tenantDomainName = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+
+            ThreadLocalProvisioningServiceProvider threadLocalServiceProvider;
+            threadLocalServiceProvider = IdentityApplicationManagementUtil
+                    .getThreadLocalProvisioningServiceProvider();
+
+            if (threadLocalServiceProvider != null) {
+
+                String serviceProvider = threadLocalServiceProvider.getServiceProviderName();
+                tenantDomainName = threadLocalServiceProvider.getTenantDomain();
+                if (threadLocalServiceProvider.getServiceProviderType() == ProvisioningServiceProviderType.OAUTH) {
+                    try {
+                        serviceProvider = ApplicationManagementService.getInstance()
+                                .getServiceProviderNameByClientId(
+                                        threadLocalServiceProvider.getServiceProviderName(),
+                                        IdentityProvisioningConstants.SP_OAUTH2_TYPE, tenantDomainName);
+                    } catch (IdentityApplicationManagementException e) {
+                        log.error("Error while provisioning", e);
+                        return true;
+                    }
+                }
+
+                // call framework method to provision the user.
+                OutboundProvisioningManager.getInstance().provision(provisioningEntity,
+                        serviceProvider, threadLocalServiceProvider.getClaimDialect(),
+                        tenantDomainName, threadLocalServiceProvider.isJustInTimeProvisioning());
+            } else {
+                // call framework method to provision the user.
+                OutboundProvisioningManager.getInstance()
+                        .provision(provisioningEntity, ApplicationConstants.LOCAL_SP,
+                                IdentityProvisioningConstants.WSO2_CARBON_DIALECT, tenantDomainName, false);
+            }
+
+            return true;
+        } finally {
+            IdentityApplicationManagementUtil.resetThreadLocalProvisioningServiceProvider();
+        }
+    }
+
+    @Override
+    public boolean doPostDeleteUserClaimValues(String userName, UserStoreManager userStoreManager)
+            throws UserStoreException {
+        return true;
+    }
+
+    @Override
+    public boolean doPreDeleteUserClaimValue(String userName, String attributeToDelete, String profileName,
+                                             UserStoreManager userStoreManager) throws UserStoreException {
+        try {
```

Comment:
Looks like you have added the same implementation logic to both pre and post listeners for deleting user claims. Any reason ? This will trigger the outbound operation twice.